tests/test_serializers/test_json.py

```diff
@@ -298,8 +301,99 @@ def test_decode_path_error():
             JsonDeserializer.decode_path(value)
 
 
+class TestUnitEncoding:
+    """ Tests for custom encoding of Unit types from pint. """
+
+    UNIT_DECODING_TEST_CASES = [
+        (Unit("meter"), CustomEncodedUnitTypes.unit.json_encoding("meter"),),
+        ("fraction", CustomEncodedUnitTypes.unit.json_encoding("fraction")),
+        ("percent", CustomEncodedUnitTypes.unit.json_encoding("percent")),
+    ]
+
+    UNIT_ENCODING_TEST_CASES = UNIT_DECODING_TEST_CASES
+
+    @staticmethod
+    def test_encode_unit_error():
+        """ Test that the JsonSerializer.encode_unit raises an error if no encoding is defined. """
+        # Arrange
+        value = "foo"
+
+        # Act & Assert
+        with pytest.raises(NotImplementedError):
+            JsonSerializer.encode_unit(value)
+
+    @staticmethod
+    @pytest.mark.parametrize("value, expected", UNIT_ENCODING_TEST_CASES)
+    def test_encode_unit(value, expected):
+        """ Test that the JsonSerializer.encode_unit encodes values as expected. """
+        # Arrange
+
+        # Act
+        result = JsonSerializer.encode_unit(value)
+
+        # Assert
+        assert result == expected
+
+    @staticmethod
+    @pytest.mark.parametrize("expected, value", UNIT_DECODING_TEST_CASES)
```

Comment:
(nit) order "value, expected" seems more common here

Reply:
oops, maybe this is intentional because this is a decoding test? subtle.

Reply:
i am just following what previous developers in this project have done